src/main/java/approximations/java/lang/StringBuffer_Tests.java

```diff
@@ -0,0 +1,2150 @@
+            } catch (StringIndexOutOfBoundsException e) {
+                return 2;
+            }
+        }
+
+        if (execution == 3) {
+            if (stringBuffer.subSequence(1, 1).equals(""))
+                return 3;
+            else
+                return -1;
+        }
+
+
+        return -1;
+    }
+
+
+    // StringBufferAutomaton::substring (int)
+    @Test(executionMax = 3)
+    public static int test_substring_0(final int execution) {
+        StringBuffer stringBuffer = new StringBuffer("0123456789");
+
+        if (execution == 0) {
+            if (stringBuffer.substring(2).equals("23456789")) return 0;
+            else return -1;
+        }
+
+
+        if (execution == 1) {
+            if (stringBuffer.substring(9).equals("9"))
+                return 1;
+            else
+                return -1;
+        }
+
+        if (execution == 2) {
+            try {
+                stringBuffer.substring(11);
+                return -1;
+            } catch (StringIndexOutOfBoundsException e) {
+                return 2;
+            }
+        }
+
+        if (execution == 3) {
+            if (stringBuffer.substring(10).equals(""))
+                return 3;
+            else
+                return -1;
+        }
+
+
+        return -1;
+    }
+
+
+    // StringBufferAutomaton::substring (int, int)
+    @Test(executionMax = 3)
+    public static int test_substring_1(final int execution) {
+        StringBuffer stringBuffer = new StringBuffer("0123456789");
+
+        if (execution == 0) {
+            if (stringBuffer.substring(2, 3).equals("2")) return 0;
+            else return -1;
+        }
+
+
+        if (execution == 1) {
+            if (stringBuffer.substring(1, 10).equals("123456789"))
+                return 1;
+            else
+                return -1;
+        }
+
+        if (execution == 2) {
+            try {
+                stringBuffer.substring(1, 11);
+                return -1;
+            } catch (StringIndexOutOfBoundsException e) {
+                return 2;
+            }
+        }
+
+        if (execution == 3) {
+            if (stringBuffer.substring(1, 1).equals(""))
+                return 3;
+            else
+                return -1;
+        }
+
+
+        return -1;
+    }
+
+
+    // StringBufferAutomaton::toString
+    @Test(executionMax = 1)
+    public static int test_toString_0(final int execution) {
+        StringBuffer stringBuffer = new StringBuffer();
+
+        if (execution == 0) {
+            if (stringBuffer.toString().equals("")) return 0;
+            else return -1;
+        }
+
+        stringBuffer.append("abcde");
+        if (execution == 1) {
+            if (stringBuffer.toString().equals("abcde"))
+                return 1;
+            else
+                return -1;
+        }
+
+
+        return -1;
+    }
+
+
+    // StringBufferAutomaton::trimToSize
+    public static int test_trimToSize_0(final int execution) {
+        return -1;
```

Comment:
Consider adding a `// FIXME: <reason>` comment here.
Also of note, the `@Test` annotation has `disabled` property - if set, the test will not be executed by the `SelfTestMain` test runner.

*Hint:* side-effects of the method-under-test can probably be captured via `capacity()` value.